PyTorch extension fails to build (permission denied error)

**Describe the bug**
The PyTorch extension fails to build on a GNU/Linux system for me (and possibly on other UNIXes), giving a permission denied error:

```
g++ -pthread -shared -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -fno-semantic-interposition -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now -L/opt/cuda/lib64 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.9/PytorchNvCodec.o -L/usr/lib/python3.9/site-packages/torch/lib -L/opt/cuda/lib64 -L/usr/lib -lc10 -ltorch -ltorch_cpu -ltorch_python -lcudart -lc10_cuda -ltorch_cuda -o /usr/bin/PytorchNvCodec.cpython-39-x86_64-linux-gnu.so
/usr/bin/ld: cannot open output file /usr/bin/PytorchNvCodec.cpython-39-x86_64-linux-gnu.so: Permission denied
collect2: error: ld returned 1 exit status
error: command '/usr/bin/g++' failed with exit code 1
make[2]: *** [CMakeFiles/PytorchNvCodec.dir/build.make:73: Pytorch_Nv_Codec] Error 1
```

This is happening because the `CMakeLists.txt` file is configured to output the PyTorch extension directly to the  `${CMAKE_INSTALL_PREFIX}/bin` directory. `CMAKE_INSTALL_PREFIX` is not typically used for placing the produced outputs, but mostly for installing files during an install stage. The default path for `CMAKE_INSTALL_PREFIX` is typically write protected on GNU/Linux systems and other UNIXes (`/usr/local`), and many package maintainers use write protected paths on it.

**To Reproduce**
```
$ git clone https://github.com/NVIDIA/VideoProcessingFramework.git
$ export CUDAToolkit_ROOT='/opt/cuda'
$ export CXXFLAGS+=' -I/opt/cuda/include'
$ export  LDFLAGS+=' -L/opt/cuda/lib64'
$ cmake -B build -S VideoProcessingFramework \
        -DCMAKE_BUILD_TYPE:STRING='None' \
        -DCMAKE_INSTALL_PREFIX:PATH='/usr' \
        -DCMAKE_CUDA_COMPILER:FILEPATH='/opt/cuda/bin/nvcc' \
        -DCMAKE_SKIP_RPATH:BOOL='YES' \
        -DGENERATE_PYTHON_BINDINGS:BOOL='TRUE' \
        -DGENERATE_PYTORCH_EXTENSION:BOOL='TRUE' \
        -DVIDEO_CODEC_SDK_INCLUDE_DIR:PATH='/usr/include/nvidia-sdk' \
        -Wno-dev
$ make -C build
```

The same error also happens when omitting `-DCMAKE_INSTALL_PREFIX`, since it defaults to the write protected path `/usr/local`.

**Expected behavior**
The build process to complete without errors, including the PyTorch extension.

**Screenshots**
Not applicable. Terminal only.

**Desktop (please complete the following information):**
 - OS: Arch Linux x86_64
 - Nvidia driver version: 495.44
 - CUDA Version: 11.4.2
 - Video Codec SDK Version: 11.1.5
 - Python Version: 3.9.7
 - python-pytorch-cuda: 1.10.0
 - Compiler: gcc 11.1.0
 - CMake: 3.21.4

**Additional context**
This can be fixed with the following patch:

```patch
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,9 +66,13 @@ if(GENERATE_PYTORCH_EXTENSION)
 	#Extension will be built using torch.utils.cpp_extension;
 	#So we just launch python script;
 	add_custom_command(	OUTPUT Pytorch_Nv_Codec
-						COMMAND cd ${PYTORCH_EXTENSION_SOURCES_DIR} && python setup.py build --build-lib="${CMAKE_INSTALL_PREFIX}/bin")
+						COMMAND cd ${PYTORCH_EXTENSION_SOURCES_DIR} && python setup.py build --build-lib="${CMAKE_CURRENT_BINARY_DIR}/PytorchNvCodec")
 
 	add_custom_target(	PytorchNvCodec
 						DEPENDS Pytorch_Nv_Codec)
 	add_dependencies(PyNvCodec PytorchNvCodec)
+	
+	find_package(Python REQUIRED)
+	execute_process(COMMAND python-config --extension-suffix OUTPUT_VARIABLE PYTHON_EXTENSION_SUFFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PytorchNvCodec/PytorchNvCodec${PYTHON_EXTENSION_SUFFIX} DESTINATION bin)
 endif(GENERATE_PYTORCH_EXTENSION)

```

If this patch is suitable, I can open a pull request.

Note: setted `DESTINATION` as `bin` to follow the other paths on `CMakeLists.txt`. A better place would be `${Python_SITEARCH}`.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,9 +66,13 @@ if(GENERATE_PYTORCH_EXTENSION)
 	#Extension will be built using torch.utils.cpp_extension;
 	#So we just launch python script;
 	add_custom_command(	OUTPUT Pytorch_Nv_Codec
-						COMMAND cd ${PYTORCH_EXTENSION_SOURCES_DIR} && python setup.py build --build-lib="${CMAKE_INSTALL_PREFIX}/bin")
+						COMMAND cd ${PYTORCH_EXTENSION_SOURCES_DIR} && python setup.py build --build-lib="${CMAKE_CURRENT_BINARY_DIR}/PytorchNvCodec")
 
 	add_custom_target(	PytorchNvCodec
 						DEPENDS Pytorch_Nv_Codec)
 	add_dependencies(PyNvCodec PytorchNvCodec)
+	
+	find_package(Python REQUIRED)
+	execute_process(COMMAND python-config --extension-suffix OUTPUT_VARIABLE PYTHON_EXTENSION_SUFFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/PytorchNvCodec/PytorchNvCodec${PYTHON_EXTENSION_SUFFIX} DESTINATION bin)
 endif(GENERATE_PYTORCH_EXTENSION)

